tst/regression/utils/compare_analytic.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+
+################################################################################
+#   Sample error functions
+#
+#   Each error function accepts the gold and test data in the format
+#   and returns a error metric
+################################################################################
+
+
+def l1_err(gold, test):
+    return np.sum(np.abs(gold - test))
+
+
+def l2_err(gold, test):
+    return np.linalg.norm(gold - test)
+
+
+def linf_err(gold, test):
+    return np.max(np.abs(gold - test))
+
+
+def l1_rel_err(gold, test):
+    mask = gold != 0
+    return np.sum(np.abs((gold[mask] - test[mask]) / gold[mask]))
+
+
+def l2_rel_err(gold, test):
+    mask = gold != 0
+    return np.linalg.norm((gold[mask] - test[mask]) / gold[mask])
+
+
+def linf_rel_err(gold, test):
+    mask = gold != 0
+    return np.max(np.abs((gold[mask] - test[mask]) / gold[mask]))
```

Comment:
Here I'm ok with only dividing by `gold` instead of `abs(gold) + abs(test)`, because we don't need to treat `gold` and `test` on equal footing.

Reply:
I would prefer the denominator be the average.

Reply:
I ended up reimplementing these error functions as a single function. I'm using the average for the denominator now.